src/azure-cli/azure/cli/command_modules/acs/_roleassignments.py

```diff
@@ -0,0 +1,168 @@
+
+from ._client_factory import (get_auth_management_client,
+                              get_graph_rbac_management_client)
+
+logger = get_logger(__name__)
+
+
+def _get_object_stubs(graph_client, assignees):
+    params = GetObjectsParameters(include_directory_object_references=True,
+                                  object_ids=assignees)
+    return list(graph_client.objects.get_objects_by_object_ids(params))
+
+
+def resolve_object_id(cli_ctx, assignee):
+    client = get_graph_rbac_management_client(cli_ctx)
+    result = None
+    if assignee is None:
+        raise AzCLIError('Inputted parameter "assignee" is None.')
+    if assignee.find('@') >= 0:  # looks like a user principal name
+        result = list(client.users.list(
+            filter="userPrincipalName eq '{}'".format(assignee)))
+    if not result:
+        result = list(client.service_principals.list(
+            filter="servicePrincipalNames/any(c:c eq '{}')".format(assignee)))
+    if not result:  # assume an object id, let us verify it
+        result = _get_object_stubs(client, [assignee])
+
+    # 2+ matches should never happen, so we only check 'no match' here
+    if not result:
+        raise AzCLIError(
+            "No matches in graph database for '{}'".format(assignee))
+
+    return result[0].object_id
+
+
+def resolve_role_id(role, scope, definitions_client):
+    role_id = None
+    try:
+        uuid.UUID(role)
+        role_id = role
+    except ValueError:
+        pass
+    if not role_id:  # retrieve role id
+        role_defs = list(definitions_client.list(
+            scope, "roleName eq '{}'".format(role)))
+        if len(role_defs) == 0:
+            raise AzCLIError("Role '{}' doesn't exist.".format(role))
+        if len(role_defs) > 1:
+            ids = [r.id for r in role_defs]
+            err = "More than one role matches the given name '{}'. Please pick a value from '{}'"
+            raise AzCLIError(err.format(role, ids))
+        role_id = role_defs[0].id
+    return role_id
+
+
+def build_role_scope(resource_group_name: str, scope: str, subscription_id: str):
+    subscription_scope = '/subscriptions/' + subscription_id
+    if scope is not None:
+        if resource_group_name:
+            err = 'Resource group "{}" is redundant because scope is supplied'
+            raise AzCLIError(err.format(resource_group_name))
+    elif resource_group_name:
+        scope = subscription_scope + '/resourceGroups/' + resource_group_name
+    else:
+        scope = subscription_scope
+    return scope
+
+
+def create_role_assignment(
+    cmd,
+    role,
+    assignee,
+    is_service_principal,
+    resource_group_name=None,
+    scope=None,
+    resolve_assignee=True,
+):
+    return _create_role_assignment(
+        cmd,
+        role,
+        assignee,
+        resource_group_name,
+        scope,
+        resolve_assignee=(is_service_principal and resolve_assignee),
+    )
+
+
+def _create_role_assignment(cmd, role, assignee,
```

Comment:
For reference, [this](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/custom.py#L1464) is the corresponding implementation in `custom.py`.